src/D2L.CodeStyle.Analyzers/Immutability/ImmutableAttributeConsistencyChecker.cs

```diff
@@ -45,7 +46,55 @@ INamedTypeSymbol typeSymbol
 			}
 		}
 
-		public void CompareConsistencyToBaseType(
+		public void CheckMethodDeclaration(
+			IMethodSymbol methodSymbol
+		) {
+			if( methodSymbol.TypeParameters.Length == 0 ) {
+				return;
+			}
+
+			MethodDeclarationSyntax methodSyntax = null;
+			Location[] parameterLocations = null;
+			Location GetLocationOfNthParameter( int N ) {
+				parameterLocations ??= new Location[ methodSymbol.TypeParameters.Length ];
+
+				if( parameterLocations[ N ] != default ) {
+					return parameterLocations[ N ];
+				}
+
+				methodSyntax ??= methodSymbol
+					.DeclaringSyntaxReferences[ 0 ]
+					.GetSyntax() as MethodDeclarationSyntax;
+
+				Location loc = parameterLocations[ N ] = methodSyntax.TypeParameterList.Parameters[ N ].GetLocation();

```

Comment:
The multiple assignments, `??=`, capturing... try to make the control flow here easier to follow please.